File: src/main/webapp/app/shared/util/email-domain-validation.ts
const isEmailDomainValid = function (value) {
  return value.endsWith('@springernature.com') || value.endsWith('@nature.com') || value.endsWith('@macmillaneducation.com');
};

export const validateEmailDomain = (email: string, ctx, input, cb) => {
  if (!email || (email !== '' && !isEmailDomainValid(email))) {
    cb("Only '@spingernature.com , @nature.com ,  @macmillaneducation.com' allowed.");
    return;
  }
  cb(true);
};

File: src/main/java/com/springernature/sndeals/service/util/EmailDomainValidation.java
package com.springernature.sndeals.service.util;

import org.apache.commons.lang3.StringUtils;

public class EmailDomainValidation
{
    public static boolean isValidEmailDomain(String email) {
        return !StringUtils.isEmpty(email) &&
            (email.endsWith("@springernature.com") || email.endsWith("@nature.com") || email.endsWith("@macmillaneducation.com"));
    }
}
